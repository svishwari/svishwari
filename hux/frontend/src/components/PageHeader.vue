<template>
  <v-card
    class="page-header--wrap d-flex justify-space-between align-center px-8 py-5"
    elevation="1"
    tile
    :color="bgColor"
    :height="headerHeight"
  >
    <div>
      <div class="page-header-container d-flex">
        <v-icon size="20" color="black">
          {{ icon }}
        </v-icon>
        <div class="header-title pl-1">
          {{ title }}
        </div>
      </div>
      <div>
        <slot name="description"></slot>
      </div>
      <slot name="left"></slot>
    </div>
    <div class="page-header--right">
      <slot name="right"></slot>
    </div>
  </v-card>
</template>

<script>
export default {
  name: "PageHeader",

  props: {
    icon: {
      type: String,
      required: false,
    },

    title: {
      type: String,
      required: false,
    },

    bgColor: {
      required: false,
      default: "white",
    },

    headerHeight: {
      type: Number,
      required: false,
      default: undefined,
    },
  },
}
</script>
<style lang="scss" scoped>
.page-header--wrap {
  .header-title {
    font-weight: 300;
    font-size: 24px;
  }
}
</style>
