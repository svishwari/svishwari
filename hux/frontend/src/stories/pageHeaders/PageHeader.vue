<template>
  <v-container class="header-container">
    <v-row class="header-title" :class="description ? '' : 'pb-4'">
      <div class="title-left text-h2">
        <icon :type="iconType" :size="40" color="primary" class="mr-2"></icon>
        <span class="mr-2">{{ title }}</span>
        <icon
          v-if="titleIcon"
          :type="titleIconSel"
          :size="40"
          color="primary"
          class="mr-2"
        ></icon>
        <icon
          v-if="maxBreadcrumbs != 'None'"
          type="Dropdown - right"
          :size="24"
          color="black"
          class="mr-2 mt-1"
        ></icon>
        <slot name="breadcrumbs" />
      </div>
      <v-spacer />
      <div class="title-right text-h3">
        <slot name="call-to-action" />
      </div>
    </v-row>
    <v-row v-if="description" class="header-description new-b1" align="center">
      {{ description }}
    </v-row>
  </v-container>
</template>

<script>
import Icon from "../icons/Icon2.vue"
export default {
  name: "PageHeader",
  components: { Icon },
  props: {
    iconType: {
      type: String,
      required: true,
    },
    title: {
      type: String,
      required: true,
      default: "Title",
    },
    description: {
      type: String,
      required: false,
    },
    titleIconSel: {
      type: String,
      required: false,
    },
    titleIcon: {
      type: Boolean,
      required: true,
    },
    maxBreadcrumbs: {
      type: String,
      required: true,
    },
  },
}
</script>

<style lang="scss" scoped>
.header-container {
  border-bottom: 1px solid #dddddd;
  color: #1e1e1e;

  .header-title {
    height: 72px;
    padding: 16px 32px 8px;
  }

  .title-left {
    display: flex;
    align-items: center;
  }

  .title-right {
    display: flex;
    align-items: center;
  }

  .header-description {
    min-height: 32px;
    padding: 0px 32px 16px;
    line-height: 24px;
    margin-top: 8px;
  }
}
</style>
