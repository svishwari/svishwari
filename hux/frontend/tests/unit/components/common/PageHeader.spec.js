import PageHeader from "@/components/PageHeader"
import { shallowMount } from "@vue/test-utils"

describe("Page Header", () => {
  test("Page header displays props content", () => {
    const props = {
      icon: "mdi-plus",
      title: "Welcome back, User",
      bgColor: "red",
    }
    const wrapper = shallowMount(PageHeader, {
      propsData: props,
    })

    expect(wrapper.text()).toContain(props.title)
    expect(wrapper.attributes("style")).toBe(
      `background-color: ${props.bgColor};`
    )
    expect(wrapper).toMatchSnapshot()
  })

  test("Page header displays slots content", () => {
    const descriptionContent = "Description"
    const leftSlotContent = "Left slot"
    const rightSlotContent = "Right slot"

    const wrapper = shallowMount(PageHeader, {
      slots: {
        description: `<div>${descriptionContent}</div>`,
        left: `<div>${leftSlotContent}</div>`,
        right: `<div>${rightSlotContent}</div>`,
      },
    })
    expect(wrapper.text()).toContain(descriptionContent)
    expect(wrapper.text()).toContain(leftSlotContent)
    expect(wrapper.text()).toContain(rightSlotContent)
    expect(wrapper).toMatchSnapshot()
  })

  test("Page header displays both slots and props content", () => {
    const descriptionContent = "Description"
    const leftSlotContent = "Left slot"
    const rightSlotContent = "Right slot"
    const props = {
      icon: "mdi-plus",
      title: "Welcome back, User",
      bgColor: "red",
    }

    const wrapper = shallowMount(PageHeader, {
      slots: {
        description: `<div>${descriptionContent}</div>`,
        left: `<div>${leftSlotContent}</div>`,
        right: `<div>${rightSlotContent}</div>`,
      },
      propsData: props,
    })
    expect(wrapper.text()).toContain(descriptionContent)
    expect(wrapper.text()).toContain(leftSlotContent)
    expect(wrapper.text()).toContain(rightSlotContent)
    expect(wrapper.text()).toContain(props.title)
    expect(wrapper.attributes("style")).toBe(
      `background-color: ${props.bgColor};`
    )
    expect(wrapper).toMatchSnapshot()
  })
})
