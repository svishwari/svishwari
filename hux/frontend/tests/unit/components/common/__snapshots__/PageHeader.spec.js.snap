// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page Header Page header displays both slots and props content 1`] = `
<v-card-stub loaderheight="4" tag="div" color="red" height="72" tile="true" flat="true" class="page-header--wrap d-flex justify-space-between align-center px-8 py-5">
  <div>
    <div class="d-flex">
      <icon-stub type="mdi-plus" size="20" color="black-darken4" variant="base" fillopacity="1" class="pr-1"></icon-stub>
      <div data-e2e="card-title" class="text-h2">
        Welcome back, User
      </div>
    </div>
    <div class="mt-2">
      <div>Description</div>
    </div>
    <div>Left slot</div>
  </div>
  <div class="page-header--right">
    <div>Right slot</div>
  </div>
</v-card-stub>
`;

exports[`Page Header Page header displays props content 1`] = `
<v-card-stub loaderheight="4" tag="div" color="red" height="72" tile="true" flat="true" class="page-header--wrap d-flex justify-space-between align-center px-8 py-5">
  <div>
    <div class="d-flex">
      <icon-stub type="mdi-plus" size="20" color="black-darken4" variant="base" fillopacity="1" class="pr-1"></icon-stub>
      <div data-e2e="card-title" class="text-h2">
        Welcome back, User
      </div>
    </div>
    <!---->
  </div>
  <div class="page-header--right"></div>
</v-card-stub>
`;

exports[`Page Header Page header displays slots content 1`] = `
<v-card-stub loaderheight="4" tag="div" color="white" height="72" tile="true" flat="true" class="page-header--wrap d-flex justify-space-between align-center px-8 py-5">
  <div>
    <div class="d-flex">
      <!---->
      <div data-e2e="card-title" class="text-h2">

      </div>
    </div>
    <div class="mt-2">
      <div>Description</div>
    </div>
    <div>Left slot</div>
  </div>
  <div class="page-header--right">
    <div>Right slot</div>
  </div>
</v-card-stub>
`;
